src/battle-dex.ts

```diff
@@ -601,45 +613,43 @@ const Dex = new class implements ModdedDex {
 
 		if (animationData[facing + 'f'] && options.gender === 'F') facing += 'f';
 		let allowAnim = !Dex.prefs('noanim') && !Dex.prefs('nogif');
-		if (allowAnim && genNum >= 6) spriteData.pixelated = false;
-		if (allowAnim && animationData[facing] && genNum >= 5) {
+		if (allowAnim && spriteData.gen >= 6) spriteData.pixelated = false;
+		if (allowAnim && animationData[facing] && spriteData.gen >= 5) {
 			if (facing.slice(-1) === 'f') name += '-f';
-			dir = gen + 'ani' + dir;
+			dir = baseDir + 'ani' + dir;
 
 			spriteData.w = animationData[facing].w;
 			spriteData.h = animationData[facing].h;
 			spriteData.url += dir + '/' + name + '.gif';
 		} else {
 			// There is no entry or enough data in pokedex-mini.js
 			// Handle these in case-by-case basis; either using BW sprites or matching the played gen.
-			if (gen === '') gen = 'gen5';
-			dir = gen + dir;
+			dir = (baseDir || 'gen5') + dir;
 
 			// Gender differences don't exist prior to Gen 4,
 			// so there are no sprites for it
-			if (genNum >= 4 && miscData['frontf'] && options.gender === 'F') {
+			if (spriteData.gen >= 4 && miscData['frontf'] && options.gender === 'F') {
 				name += '-f';
 			}
 
 			spriteData.url += dir + '/' + name + '.png';
 		}
 
 		if (!options.noScale) {
-			if (fieldGenNum > 5) {
+			if (graphicsGen > 4) { // TODO: should this be spriteData.gen?
 				// no scaling
-			} else if (!spriteData.isBackSprite || fieldGenNum === 5) {
+			} else if (!spriteData.isBackSprite) {
 				spriteData.w *= 2;
 				spriteData.h *= 2;
 				spriteData.y += -16;
 			} else {
-				// backsprites are multiplied 1.5x by the 3D engine
+				// old gen backsprites are multiplied by 1.5x by the 3D engine
 				spriteData.w *= 2 / 1.5;
 				spriteData.h *= 2 / 1.5;
 				spriteData.y += -11;
 			}
-			if (fieldGenNum === 5) spriteData.y = -35;
-			if (fieldGenNum === 5 && spriteData.isBackSprite) spriteData.y += 40;
-			if (genNum <= 2) spriteData.y += 2;
+			if (spriteData.gen <= 2) spriteData.y += 2;
+			if (spriteData.gen === 5 && spriteData.isBackSprite) spriteData.y += 40; // TODO: determine correct offset
```

Comment:
as above, i need to test this given we're still planning to scale the backsprite